quinn-proto/src/config.rs

```diff
@@ -321,6 +324,28 @@ impl TransportConfig {
         self.enable_segmentation_offload = enabled;
         self
     }
+
+    /// Whether to send observed address reports to peers.
+    ///
+    /// This will aid peers in inferring their reachable address, which in most NATd networks
+    /// will not be easily available to them.
+    pub fn report_observed_addresses_to_peers(&mut self, enabled: bool) -> &mut Self {
+        self.address_discovery_role
+            .provide_reports_to_peers(enabled);
+        self
+    }
+
+    /// Whether to accept observed address_reports from other peers.
+    ///
+    /// Peers with the address discovery extension enabled that are willing to provide observed
+    /// address reports will do so if this transport parameter is set. In general, observed address
+    /// reports cannot be trusted. This, however, can aid the current endpoint in inferring its
+    /// reachable address, which in most NATd networks will not be easily available.
+    pub fn accept_observed_address_reports(&mut self, enabled: bool) -> &mut Self {
```

Comment:
I see your point but I find this more clear because "accept observed addresses" can be understood as "trust them" which we intentionally do not tackle, and it's up to the application to decide. I'm going to change the `report` one to `send_.._report` instead

Reply:
everything is now sending and receiving reports, no more accept, provide etc. Naming is not the most beautiful, but it's consistent and clear now